frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -185,6 +185,54 @@ export class FieldValuesWidget extends Component {
     }
   }
 
+  getSearchableTokenFieldPlaceholder(fields, firstField) {
+    let placeholder;
+
+    const names = new Set(
+      fields.map(field => stripId(this.searchField(field).display_name)),
+    );
+
+    if (names.size > 1) {
+      placeholder = t`Search`;
+    } else {
+      const [name] = names;
+
+      placeholder = t`Search by ${name}`;
+      if (firstField.isID() && firstField !== this.searchField(firstField)) {
+        placeholder += t` or enter an ID`;
+      }
+    }
+    return placeholder;
+  }
+
+  getNonSearchableTokenFieldPlaceholder(firstField) {
+    if (firstField.isID()) {
+      return t`Enter an ID`;
+    } else if (firstField.isNumeric()) {
+      return t`Enter a number`;
+    } else {
+      return t`Enter some text`;
+    }
+  }
```

Comment:
I'd much prefer this notation in this case…5 lines instead of 8.
Linter does not allow it.

```suggestion
  getNonSearchableTokenFieldPlaceholder(firstField) {
    if (firstField.isID()) return t`Enter an ID`;
    if (firstField.isNumeric()) return t`Enter a number`;
    return t`Enter some text`;
  }
```